src/tokens/token-groups/legacy-tokens.json

```diff
@@ -1,7 +1,9 @@
 {
   "override-loading-z-index": "514",
   "choice-size": "20px",
-  "icon-size": "10px",
+  "icon-size-small": "10px",
```

Comment:
I would be inclined to move these out of `legacy-tokens` and into `\tokens`.

Reply:
This is tough. Long term I would prefer these to use values from the spacing tokens. However that would affect all the icons designs. I don't mind it staying in legacy as a follow up. It doesn't feel like a token we want to scale.

Reply:
I agree, but couldn't decide the appropriate/existing token group to add it to or if I should create a new one. Was hoping to iron that out here in the PR. Any name suggestions? I considered `token-groups/sizes.json`, but think that may be more appropriate for a token scale. Another option is `token-groups/icons.json`.

Reply:
I would leave it in legacy as it feels like something we want to adjust long term.

Reply:
> We also should move away from `small` `medium` and `large` as it makes adding or removing values awkward.
> I would prefer icon-10 and icon-20 for now in legacy. Happy for other decisions though 🤷‍♂️

I'm actually not opposed to `small`, `medium`, `large`, etc. sizes for `icons`, `buttons`, `badges`, etc.
That convention obviously doesn't work for token scales for the reason you've outlined (`makes adding or removing values awkward`), but makes perfect sense in the context of theming and limiting the variance on a page.

References:
<details>
  <summary>Design system references:</summary>

Primer
![image](https://user-images.githubusercontent.com/32409546/151889678-2f2d850f-a915-488a-a2c0-37a421f47a6b.png)

MUI
![image](https://user-images.githubusercontent.com/32409546/151889775-35efd8ec-ea4f-48a4-aacd-e4550382efb3.png)

Spectrum
![image](https://user-images.githubusercontent.com/32409546/151889829-e7d7767b-c6ff-48c9-8c32-60e6b7e85260.png)
</details>

Reply:
It also adds a layer of abstraction. As a user I don't know if `small` is `10px` `12px` `14px`. I have to either read documentation or go into code.

Reply:
This system should in the future have a consistent grid. Elements should snap to a 4px value. My concerns:

- Having values like `10px` and `20px` instantly breaks this rule
- We have to document and make sure users understand multiple tokens (spacing system, element widths)
- There should be one approach to sizes/spacing/layout
- `sm` `md` `lg` abstracts the values and makes it harder to understand without reading documentation or inspecting code

The reality is that the system doesn't do this. Now is the wrong time to do this work but it's important to make sure we aren't making this harder in the future. For this PR I would recommend shipping `sm` `md` and removing `lg`. This allows us to change the values and iterate towards a consolidated system for layout.